resolwe_bio/processes/support_processors/multiqc.py

```diff
@@ -0,0 +1,242 @@
+                    create_symlink(d.stats.path, os.path.join(sample_dir, rrna_report))
+                elif d.build == 'globin':
+                    globin_report = f'{bam_name}.globin.Log.final.out'
+                    create_symlink(d.stats.path, os.path.join(sample_dir, globin_report))
+                else:
+                    report = f'{bam_name}.Log.final.out'
+                    create_symlink(d.stats.path, os.path.join(sample_dir, report))
+
+            elif d.type.startswith('data:alignment:bam'):
+                name = os.path.basename(d.stats.path)
+                create_symlink(d.stats.path, os.path.join(sample_dir, name))
+
+            elif d.type == 'data:expression:featurecounts:':
+                name = os.path.basename(d.counts_summary.path)
+                create_symlink(d.counts_summary.path, os.path.join(sample_dir, name))
+
+            elif d.type == 'data:chipseq:callpeak:macs2:':
+                name = os.path.basename(d.called_peaks.path)
+                create_symlink(d.called_peaks.path, os.path.join(sample_dir, name))
+
+            elif d.type == 'data:samtools:idxstats:':
+                name = os.path.basename(d.report.path)
+                create_symlink(d.report.path, os.path.join(sample_dir, name))
+
+            elif d.type == 'data:qorts:qc:':
+                qorts_path = os.path.join(sample_dir, 'QoRTs')
+                os.mkdir(qorts_path)
+                name = os.path.basename(d.summary.path)
+                create_symlink(d.summary.path, os.path.join(qorts_path, name))
+
+            elif d.type == 'data:expression:salmon:':
+                create_symlink(d.salmon_output.path, sample_dir)
+
+            elif d.type.startswith('data:alleyoop'):
+                create_symlink(d.report.path, sample_dir)
+                # Alleyoop summary may contain PCA plot data
+                if os.path.isfile(d.plot_data.path):
+                    create_symlink(d.plot_data.path, sample_dir)
+
+            else:
+                unsupported_data.append(d.id)
+
+        if unsupported_data:
+            self.warning(f'Data with ID(s) {unsupported_data} is not supported by the MultiQC analysis.')
+
+        create_summary_table(samples, species, build)
+
+        args = [
+            '-dd', inputs.advanced.dirs_depth,
+        ]
+
+        if inputs.advanced.dirs:
+            args.append('-d')
+
+        if inputs.advanced.fullnames:
+            args.append('-s')
+
+        if inputs.advanced.config:
+            args.extend(['-c', '/opt/resolwebio/assets/multiqc_config.yml'])
+
+        if inputs.advanced.cl_config:
+            args.extend(['--cl-config ', inputs.advanced.cl_config])
+
+        with Cmd.env(LC_ALL='C.UTF-8'):
+            return_code, _, _ = Cmd['multiqc']['.'][args] & TEE(retcode=None)
+            if return_code:
+                self.error('MultiQC analysis failed.')
+
+        if os.path.isdir('multiqc_data') and os.path.isfile('multiqc_report.html'):
```

Comment:
I would make outputs output in any case, but check for existence of folders/files and fail if they do not exist.

```
if not os.path.isdir('multiqc_data') and not os.path.isfile('multiqc_report.html'):
    self.error('MultiQC finished without creating outputs.')

outputs.report = 'multiqc_report.html'
outputs.report_data = 'multiqc_data'
            
```